libnetwork/osl/interface_linux_test.go

```diff
@@ -0,0 +1,94 @@
+package osl
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/docker/docker/internal/nlwrap"
+	"github.com/docker/docker/internal/sliceutil"
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netns"
+	"gotest.tools/v3/assert"
+)
+
+func TestGenerateIfaceName(t *testing.T) {
+	testcases := []struct {
+		prefix string
+		names  []string
+		want   string
+	}{
+		{prefix: "test", names: []string{"test0", "test1"}, want: "test2"},
+		{prefix: "test", names: []string{"test0", "test2"}, want: "test1"},
+		{prefix: "test", names: []string{"test2"}, want: "test0"},
```

Comment:
See my other comment; not sure how realistic it is to happen, but we should probably have test-cases with mixed prefixes, and one that has a `-` in the prefix; for example, this will fail, and pick `eth0`;


```suggestion
		{prefix: "test", names: []string{"test2"}, want: "test0"},
		{prefix: "eth", names: []string{"eth-0", "eth-1", "eth0"}, want: "eth1"},
```

Reply:
Done - added a test case for negative numbers.

There was an edge case around `-0` being treated as `0` - it's now handled correctly.